### PR TITLE
refactor: tap-core upgrade

### PR DESCRIPTION
**Pull request by gusinacio**

### CODE REVIEW

File: common/src/indexer_service/http/indexer_service.rs [hunk 1]
```diff
@@ -279,19 +280,25 @@ impl IndexerService {
...
             .await?;
 
-        let tap_manager = TapManager::new(
+        let domain_separator = eip712_domain! {
+            name: "TAP",
+            version: "1",
+            chain_id: options.config.scalar.chain_id,
+            verifying_contract: options.config.scalar.receipts_verifier_address,
+        };
+        let indexer_executor =
```
Comment (aasseman) [inline common/src/indexer_service/http/indexer_service.rs:289]:
```suggestion
        let indexer_context =
```

**Review by aasseman — CHANGES_REQUESTED** [review]
Just a few little things.
I may have missed places where `context` is still called `executor`.

File: common/src/tap/checks/deny_list_check.rs [hunk 2]
```diff
@@ -1,43 +1,35 @@
...
-pub struct TapManager {
-    indexer_allocations: Eventual<HashMap<Address, Allocation>>,
+pub struct DenyListCheck {
     escrow_accounts: Eventual<EscrowAccounts>,
-    pgpool: PgPool,
-    domain_separator: Arc<Eip712Domain>,
+    domain_separator: Eip712Domain,
     sender_denylist: Arc<RwLock<HashSet<Address>>>,
-    _sender_denylist_watcher_handle: Arc<tokio::task::JoinHandle<()>>,
+    _sender_denylist_watcher_handle: Option<Arc<tokio::task::JoinHandle<()>>>,
```
[the remark this answers is not in the document]
Comment (gusinacio) [reply common/src/tap/checks/deny_list_check.rs:23]: Indeed, I was refactoring to use something else where I needed the Option, but in the end, it wasn't needed 😅

File: common/src/tap/checks/deny_list_check.rs [hunk 3]
```diff
@@ -1,43 +1,35 @@
...
+impl DenyListCheck {
     pub async fn new(
         pgpool: PgPool,
-        indexer_allocations: Eventual<HashMap<Address, Allocation>>,
         escrow_accounts: Eventual<EscrowAccounts>,
         domain_separator: Eip712Domain,
     ) -> Self {
-        // Listen to pg_notify events. We start it before updating the sender_denylist so that we
-        // don't miss any updates. PG will buffer the notifications until we start consuming them.
         let mut pglistener = PgListener::connect_with(&pgpool.clone()).await.unwrap();
```
[the remark this answers is not in the document]
Comment (gusinacio) [reply common/src/tap/checks/deny_list_check.rs:33]: I created this whole file from scratch and started copying the checks one by one. Git thinks that this is the same file as `tap_manager.rs`. That's why it looks like I intentionally removed it. Gonna add it back.

**Pull request closed without merging**